general/development/tools/phpunit/upgrading-11.md

```diff
@@ -0,0 +1,198 @@
+---
+title: PHPUnit 11 Upgrade
+tags:
+- Testing
+- PHPUnit
+---
+
+<Since version="5.0" issueNumber="MDL-83468" />
+
+Moodle 5.0 updated the version of PHPUnit from version 9.6 to version 11.
+
+This guide is intended to help you update your tests to work with the newer versions.
+
+## Why did we make this change?
+
+PHPUnit 9.6 is currently in maintenance support, but we don't know for how much longer.
+
+We ultimately _need_ to update to a newer version of PHPUnit at some point to ensure that PHP version incompatibilities are addressed and we are able to keep testing Moodle properly.
+
+The most appropriate target is for Moodle 5.0 which follows our new Series naming and versioning policies.
+
+### Why not PHPUnit 10?
+
+- PHPUnit 10 is the next major version of PHPUnit and introduces a number of compatibility issues with PHPUnit 9.6.
+- PHPUnit 11 is the latest major version of PHPUnit and is _more_ backwards compatible with PHPunit 9.6 than PHPUnit 10 is.
+
+Many of the features removed in PHPUnit 10 were then partially added back in PHPUnit 11.
+
+To give a specific example, PHPUnit 9.6 has a `--debug` flag. This was removed in 10.0, and added back in 11.0 as an alias for a combination of other flags.
+
+There are lots of other examples of this kind of thing.
+
+## Can I make my tests work in both 9.6 and 11.4?
+
+Yes, _usually_. But it may require some changes to your tests and/or code. For the most part any such changes are a good thing and in all cases I've encountered they've been because the tests were wrong in the first place anyway.
```

Comment:
Would the "I" in `I've encountered` sound confusing to the reader, given that authors are not indicated in the dev docs pages (only the person who last updated the page)? Would something like `we've encountered` be more appropriate?